In [28]:
# Import necessary libraries
import pandas as pd
import numpy as np
from scipy.stats import ttest_1samp
from scipy import stats
from statsmodels.stats import weightstats as stests

In [29]:
# Read the Pokemon dataset into a DataFrame
df = pd.read_csv('pokemon.csv')
df.head()

#                   Name Type 1  Type 2  Total  HP  Attack  Defense  \
0  1              Bulbasaur  Grass  Poison    318  45      49       49   
1  2                Ivysaur  Grass  Poison    405  60      62       63   
2  3               Venusaur  Grass  Poison    525  80      82       83   
3  3  VenusaurMega Venusaur  Grass  Poison    625  80     100      123   
4  4             Charmander   Fire     NaN    309  39      52       43   

   Sp. Atk  Sp. Def  Speed  Generation  Legendary  
0       65       65     45           1      False  
1       80       80     60           1      False  
2      100      100     80           1      False  
3      122      120     80           1      False  
4       60       50     65           1      False

In [30]:
# Random Sampling 
def random_sampling(df,no_of_samples, sample_size):
    random_samples = []   
    for i in range(no_of_samples):
        # Randomly sampling without replacement
        sample = np.random.choice(df, size=sample_size, replace=False)
        random_samples.append(sample)
    random_samples = np.array(random_samples)
    return random_samples
pokemon_samples = random_sampling(df['Total'], 200, 50)
pokemon_samples

array([[630, 334, 480, ..., 600, 340, 580],
       [260, 610, 302, ..., 455, 335, 300],
       [487, 440, 475, ..., 413, 600, 505],
       ...,
       [495, 465, 470, ..., 525, 430, 395],
       [424, 320, 300, ..., 300, 325, 567],
       [360, 290, 270, ..., 523, 494, 300]])

In [31]:
#P-value
attack_mean = np.mean(df['Attack'])
print("mean = ", attack_mean)
test, pval = ttest_1samp(df['Attack'], 73)
print("p-values = ", pval)
if pval < 0.05:
   print("Reject null hypothesis")
else:
  print("Fail to reject null hypothesis")

mean =  79.00125
p-values =  2.1705920330696673e-07
Reject null hypothesis


### One-sample t-test for Attack attribute:

Null Hypothesis: The average Attack of Pokémon is 73.

Alternate Hypothesis: The average Attack of Pokémon is not 73.

Result: With a very low p-value (2.17e-07), we reject the null hypothesis. This indicates strong evidence that the average Attack of Pokémon is significantly different from 73.

In [32]:
# Z-Test 
ztest ,pval = stests.ztest(df['Defense'], x2=None, value=73)
print(float(pval))
if pval<0.05:
    print("Reject null hypothesis")
else:
    print("Fail to reject null hypothesis")

0.4447658863352716
Fail to reject null hypothesis


### Z-test for Defense attribute:

Null Hypothesis: The average Defense of Pokémon is 73.

Alternate Hypothesis: The average Defense of Pokémon is not 73.

Result: The p-value (0.4448) is greater than the significance level of 0.05. Therefore, we fail to reject the null hypothesis. This suggests that there isn't sufficient evidence to conclude that the average Defense of Pokémon differs significantly from 73.

### Conclusion 
There is strong evidence to suggest that the population mean Attack value differs significantly from 73, while there is no significant difference between the population mean Defense value and 73.